instrumentation/lettuce/lettuce-5.1/src/main/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_1/RedisCommandNormalizer.java

```diff
@@ -0,0 +1,208 @@
+    normalizers.put("RPUSHX", keepOneArg);
+
+    // Pub/Sub
+    normalizers.put("PUBLISH", keepOneArg);
+
+    // Scripting
+    normalizers.put("EVAL", Eval.INSTANCE);
+    normalizers.put("EVALSHA", Eval.INSTANCE);
+
+    // Sets
+    normalizers.put("SADD", keepOneArg);
+    normalizers.put("SISMEMBER", keepOneArg);
+    normalizers.put("SMISMEMBER", keepOneArg);
+    normalizers.put("SMOVE", keepTwoArgs);
+    normalizers.put("SREM", keepOneArg);
+
+    // Sorted Sets
+    normalizers.put("ZADD", keepOneArg);
+    normalizers.put("ZCOUNT", keepOneArg);
+    normalizers.put("ZINCRBY", keepOneArg);
+    normalizers.put("ZLEXCOUNT", keepOneArg);
+    normalizers.put("ZMSCORE", keepOneArg);
+    normalizers.put("ZRANGEBYLEX", keepOneArg);
+    normalizers.put("ZRANGEBYSCORE", keepOneArg);
+    normalizers.put("ZRANK", keepOneArg);
+    normalizers.put("ZREM", keepOneArg);
+    normalizers.put("ZREMRANGEBYLEX", keepOneArg);
+    normalizers.put("ZREMRANGEBYSCORE", keepOneArg);
+    normalizers.put("ZREVRANGEBYLEX", keepOneArg);
+    normalizers.put("ZREVRANGEBYSCORE", keepOneArg);
+    normalizers.put("ZREVRANK", keepOneArg);
+    normalizers.put("ZSCORE", keepOneArg);
+
+    // Streams
+    normalizers.put("XADD", new MultiKeyValue(2));
+
+    // Strings
+    normalizers.put("APPEND", keepOneArg);
+    normalizers.put("GETSET", keepOneArg);
+    normalizers.put("MSET", setMultiField);
+    normalizers.put("MSETNX", setMultiField);
+    normalizers.put("PSETEX", keepTwoArgs);
+    normalizers.put("SET", keepOneArg);
+    normalizers.put("SETEX", keepTwoArgs);
+    normalizers.put("SETNX", keepOneArg);
+    normalizers.put("SETRANGE", keepOneArg);
+
+    NORMALIZERS = unmodifiableMap(normalizers);
+  }
+
+  public static String normalize(String command, List<String> args) {
+    return NORMALIZERS
+        .getOrDefault(command.toUpperCase(), Default.INSTANCE)
+        .normalize(command, args);
+  }
+
+  public interface CommandNormalizer {
+    String normalize(String command, List<String> args);
+
+    enum Default implements CommandNormalizer {
+      INSTANCE;
+
+      @Override
+      public String normalize(String command, List<String> args) {
+        StringBuilder normalised = new StringBuilder(command);
+        for (String arg : args) {
+          normalised.append(" ").append(arg);
+        }
+        return normalised.toString();
+      }
+    }
+
+    class CommandAndNumArgs implements CommandNormalizer {
+      private final int numOfArgsToKeep;
+
+      public CommandAndNumArgs(int numOfArgsToKeep) {
+        this.numOfArgsToKeep = numOfArgsToKeep;
+      }
+
+      @Override
+      public String normalize(String command, List<String> args) {
+        StringBuilder normalised = new StringBuilder(command);
+        int i = 0;
+        for (; i < numOfArgsToKeep && i < args.size(); ++i) {
+          normalised.append(" ").append(args.get(i));
+        }
+        for (; i < args.size(); ++i) {
```

Comment:
I think this is easier to read, but curious other thoughts, I also see attractiveness of current approach which increments a single counter to show that everything is hit

```suggestion
        for (int i = 0; i < numOfArgsToKeep && i < args.size(); ++i) {
          normalised.append(" ").append(args.get(i));
        }
        for (int i = numOfArgsToKeep; i < args.size(); ++i) {
```

Reply:
Both are perfectly fine to me, I'll use your version if you think it's more readable.